src/prof/src/lib.rs

```diff
@@ -223,6 +226,26 @@ mz_fg_version: 1
                 build_id: build_id_idx,
                 ..Default::default()
             });
+
+            // This is a is a Polar Signals-specific extension: For correct offline symbolization
+            // they need access to the memory offset of mappings, but the pprof format only has a
+            // field for the file offset. So we instead encode additional information about
+            // mappings in magic comments. There must be exactly one comment for each mapping.
+
+            // Take a shortcut and assume the ELF type is always `ET_DYN`. This is true for shared
+            // libraries and for position-independent executable, so it should always be true for
+            // any mappings we have.
+            // Getting the actual information is annoying. It's in the ELF header (the `e_type`
+            // field), but there is no guarantee that the full ELF header gets mapped, so we might
+            // not be able to find it in memory. We could try to load it from disk instead, but
+            // then we'd have to worry about blocking disk I/O.
```

Comment:
Note that if it is actually mapped, I don't think the I/O will block, as the page will be in the filesystem cache, if I understand correctly how Linux works.

(Btw, I'm not arguing that we should actually try to do file i/o here... this hack is fine)